### ISSUE TITLE
*Low Priority* Tests seem significantly slower on Firefox than Safari or Chrome

### ISSUE THREAD

**Issue by kaspnilsson** [Polymer/web-component-tester#465]
I'm a member of a project with a relatively large number of tests in WCT (>1000) and I notice that our full suite takes about twice as long to execute in Firefox than Safari and Chrome. I'm mostly curious as to why this is, and if I can do something with my FF configuration or something to make it faster.

Firefox v50, Chrome v55, Safari v10.

Thanks!

**Comment by rictic** [comment]
Do you have a large number of wct suites? (i.e. a large number of files containing tests?) It's possible that something about the way we create and initialize iframes is slow on Firefox.

**Comment by kaspnilsson** [comment]
Yes, we have 74 individual files, most of which have multiple suites within.

**Comment by rictic** [comment]
I'd give ~40% odds that the discrepancy is in your code, and ~40% that it's in our suite initialization code, and ~20% odds that it's in the in-suite test runner code (e.g. mocha).

A few ways to get more info:
* Try smashing all of your tests into one giant file (you probably don't want to check in such a change, just for debug)
* Make a set of really simple `expect(1).to.be.equal(1)` tests spread over a large number of files.
* Run your tests in mocha without WCT somehow? Not an obvious way to do this, but probably possible.

**Comment by dalejoel** [comment]
I run a test suite of 4 files with a total of 19 tests that get run twice because we test against the shadow dom as well. That's not a big run.
Chrome finishes in 6 seconds.
Firefox takes 55 seconds.

**Comment by rictic** [comment]
@dalejoel Do you see a similar performance difference for the code under test, or is it only through WCT that you see this performance difference?

**Comment by dalejoel** [comment]
I don't understand the former part of your question. I just know that when I run WCT tests, we can get our Chrome tests to only take 5 seconds but Firefox takes 60 seconds. That is a ridiculously large difference.

**Comment by rictic** [comment]
What I'm asking is, is this from WCT or from your code? Like, when you run your code normally (in production or whatever) is Firefox far slower than Chrome? I mean, there's a lot of ways to write code that's way faster in one browser than another.

**Comment by dalejoel** [comment]
Oh. Gotcha. I don't know. While I can say Firefox has at times seemed less
happy with polymer, normal usage in the browser is not something that jumps
out because I'm not slamming a ton of UI components like in the unit tests.
So the differences might seem smaller. But I have noticed some rendering
differences in terms of timing of rendering.
